cmd/deployIntegrationArtifact.go

```diff
@@ -74,12 +74,11 @@ func runDeployIntegrationArtifact(config *deployIntegrationArtifactOptions, tele
 	httpClient.SetOptions(clientOptions)
 
 	deployResp, httpErr := httpClient.SendRequest("POST", deployURL, nil, header, nil)
+	if httpErr != nil {
+		return errors.Wrapf(httpErr, "HTTP %v request to %v failed with error", "POST", deployURL)
+	}
 
-	defer func() {
-		if deployResp != nil && deployResp.Body != nil {
-			deployResp.Body.Close()
-		}
-	}()
+	defer deployResp.Body.Close()
 
 	if deployResp == nil {
 		return errors.Errorf("did not retrieve a HTTP response: %v", httpErr)
```

Comment:
With the change of checking `httpErr != nil` above, `httpErr` will definitely be `nil` here! How about using `errors.Error` to create a new error.